### ISSUE TITLE
[BUG] Java should be compiled with parameters=true

### ISSUE THREAD

**Issue by kpavlov** [langchain4j/langchain4j#2107]
**Describe the bug**
When javac `parameters` is set to true, then parameters information is available at runtime. It allows to make better error mesages, e.g. `"Parameter 'userMessage' of method 'illegalChat3' should be annotated with @v or @Usermessage or @UserName or @memoryid"` instead of `"Parameter 'arg0' of method 'illegalChat3' should be annotated with @v or @Usermessage or @UserName or @memoryid"`  (see AiServicesUserMessageConfigTest.test_illegal_user_message_configuration_3: langchain4j/src/test/java/dev/langchain4j/service/AiServicesUserMessageConfigTest.java#L232).

**Todo**
1. Enable parameters, cleanup redundant compiler plugin settings, see #2105
2. Fix tests: replace `argN` with parameter names, [example](https://github.com/langchain4j/langchain4j/pull/2105/commits/d39ca237a4610b3cd2bceb484ae023fa7f0ed556)

**Log and Stack trace**
 [error](https://github.com/langchain4j/langchain4j/actions/runs/11824379205/job/32945750912)

**To Reproduce**
[set parameters=true](https://github.com/langchain4j/langchain4j/pull/2105/commits/1e581a2a3c1d92e4060ab2760ba159ca60966fff#diff-db1dfc1ccffac6c8f8ae2414e6aad535940aaf08a488d30941157dc53ec573b0R394)

**Expected behavior**
Code is compiled with `parameters=true`

**Please complete the following information:**
- LangChain4j version: 0.36.0
- LLM(s) used: core functionality
- Java version: 17,21,23
- Spring Boot version (if applicable): 

**Additional context**
Discovered when `parameters=true` was set [here](https://github.com/langchain4j/langchain4j/pull/2105/commits/1e581a2a3c1d92e4060ab2760ba159ca60966fff) / [error](https://github.com/langchain4j/langchain4j/actions/runs/11824379205/job/32945750912)

**Comment by langchain4j** [comment]
@kpavlov `parameters` are kept disabled because this is the default setting and many users will use LC4j with this setting.
If we'll change it for our ITs, it might mask issues with prompt templates, tools and structured outputs.

**Comment by w4tson** [comment]
Just been working with llama3.2 and it suffers badly when deciding to call tools if this flag is not enabled. More powerful models can cope with arguments names as "arg0" etc. But this is a heck of a gotcha, I think it could be spelled out more in either the docs or examples

**Comment by dliubarskyi** [comment]
Hi @w4tson, thanks for the feedback! What framework are you using?

**Comment by w4tson** [comment]
I am using SpringBoot + Gradle of course! I have an allergy to even small amounts of XML. 

Confession Time: I was using Kotlin.

I've dug deeper on this and it must be a Kotlin Compiler default parameter thing. The example in the repo works fine when using Java+SpringBoot+Gradle. The reflection API gets the parameter names just fine.

So It's not a big a gotcha as I thought.

**Comment by kpavlov** [comment]
Kotlin code is [already compiled](https://github.com/langchain4j/langchain4j/blob/b0b6d91385d87d044420e0ab67ed91c37f28503c/langchain4j-parent/pom.xml#L544) with `javaParameters=true`, but java code doesn't. 

> [`-parameters`](https://docs.oracle.com/en/java/javase/17/docs/specs/man/javac.html)
> Generates metadata for reflection on method parameters. Stores formal parameter names of constructors and methods in the generated class file so that the method java.lang.reflect.Executable.getParameters from the Reflection API can retrieve them.

Enabling parameters in java should only improve user experience bu adding metadata to LC4j classes, although it's a breaking change.

**Comment by dliubarskyi** [comment]
@kpavlov just in case, enabling parameters option on our side makes no effect on the compilation of the user code. If user wants to have parameter names of his custom POJOs, they need to make sure they compile their code with parameters option.

**Comment by kpavlov** [comment]
Yes, but don't we want to encourage users to do so and show the example?
If not, let's just close this issue

**Comment by dliubarskyi** [comment]
@kpavlov we should explicitly mention `parameters` compiler option in the documentation, but I doubt enabling it in our repo will make any difference for users
